sql/catalyst/src/main/scala/org/apache/spark/sql/metricview/serde/MetricViewCanonical.scala

```diff
@@ -0,0 +1,184 @@
+
+  override def toString: String = this.name
+}
+
+// SQL source, representing a SQL query
+private[sql] case class SQLSource(sql: String) extends Source {
+  val sourceType: SourceType = SourceType.SQL
+
+  override def toString: String = this.sql
+}
+
+private[sql] object Source {
+  def apply(sourceText: String): Source = {
+    if (sourceText.isEmpty) {
+      throw MetricViewValidationException("Source cannot be empty")
+    }
+    Try(CatalystSqlParser.parseTableIdentifier(sourceText)) match {
+      case Success(_) => AssetSource(sourceText)
+      case Failure(_) =>
+        Try(CatalystSqlParser.parseQuery(sourceText)) match {
+          case Success(_) => SQLSource(sourceText)
+          case Failure(queryEx) =>
+            throw MetricViewValidationException(
+              s"Invalid source: $sourceText",
+              Some(queryEx)
+            )
+        }
+    }
+  }
+}
+
+private[sql] case class Column(
+    name: String,
+    expression: Expression,
+    ordinal: Int) {
+  def columnType: ColumnType = expression match {
+    case _: DimensionExpression => ColumnType.Dimension
+    case _: MeasureExpression => ColumnType.Measure
+    case _ =>
+      throw MetricViewValidationException(
+        s"Unsupported expression type: ${expression.getClass.getName}"
+      )
+  }
+
+  def getColumnMetadata: ColumnMetadata = {
+    val expr = expression.expr
+    if (expr.length > Constants.MAXIMUM_PROPERTY_SIZE) {
+      throw MetricViewValidationException(
+        s"Expression length ${expr.length} exceeds maximum allowed size " +
+        s"${Constants.MAXIMUM_PROPERTY_SIZE} for column '$name'"
+      )
+    }
+    ColumnMetadata(columnType.toString, expr)
+  }
+}
+
+private[sql] object ColumnType extends Enumeration {
+  type ColumnType = Value
+  val Dimension: ColumnType = Value("dimension")
+  val Measure: ColumnType = Value("measure")
+
+  // Method to match case-insensitively and return the correct value
+  def fromString(columnType: String): ColumnType = {
+    values.find(_.toString.equalsIgnoreCase(columnType)).getOrElse {
+      throw MetricViewFromProtoException(
+        s"Unsupported column type: $columnType"
+      )
+    }
+  }
+}
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(Include.NON_ABSENT)
+private[sql] case class ColumnMetadata(
+    @JsonProperty(value = Constants.COLUMN_TYPE_PROPERTY_KEY, required = true)
+    columnType: String, // "type" -> "metric_view.type"
+    @JsonProperty(value = Constants.COLUMN_EXPR_PROPERTY_KEY, required = true)
+    expr: String // "expr" -> "metric_view.expr"
+)
+
+// Only parse the "version" field and ignore all others
+@JsonIgnoreProperties(ignoreUnknown = true)
+private[sql] case class YAMLVersion(version: String) {
+  private def validYAMLVersions: Set[String] = Set("0.1")
+}
+
+private[sql] object YAMLVersion {
+  private def validYAMLVersions: Set[String] = Set("0.1")
+
+  def apply(version: String): YAMLVersion = {
```

Comment:
dead code for now?

Reply:
good catch. the actual version validation is at MetricViewFactory and this code is useless. There is also a test that confirm this: https://github.com/apache/spark/pull/53146/files#diff-23bf5ddc582ff6684f7cc8950a12f4d8e745ff3fb7b0142dd00015e1f159fc8aR144